src/main/java/bb/example/UserInterface.java

```diff
@@ -93,91 +103,77 @@ private String enterPhoneNumber() {
 
-            if (doesContainRangeOfDigits(age)) {
+            if (Validators.doesContainRangeOfDigits(age)) {
                 return age;
+            } else {
+                outputHandler.print("The age must be in the range of 1-100");
             }
         }
     }
 
-    private boolean doesContainRangeOfDigits(int age) {
-        if (!(age > 0 && age <= 100)) {
-            System.out.println("The age must be in the range of 1-100");
-            return false;
-        }
-        return true;
-    }
 
     private String displayAllInsured() {
-        Collection<PersonI> persons = database.listOfAllPersons();
+        Collection<Person> persons = database.listOfAllPersons();
         if (persons.isEmpty()) {
             return "No insured individuals are recorded in the database";
         }
-        StringBuilder result = new StringBuilder();
-        for (PersonI person : persons) {
-            result.append(person).append("\n");
-        }
-        return result.toString();
+
+        return Utilities.buildPersonsList(persons);
     }
 
     private String displayInsured() {
 
         System.out.println("Enter the name or surname:");
-        String inputNameSurname = scanner.nextLine().trim();
+        String inputNameSurname = inputHandler.getInput();
 
-        Collection<PersonI> foundPersons = database.findSpecificPerson(inputNameSurname);
-        StringBuilder result = new StringBuilder();
-        for (PersonI person : foundPersons) {
-            result.append(person).append("\n");
-        }
+        Collection<Person> persons = database.findSpecificPerson(inputNameSurname);
 
-        if (foundPersons.isEmpty()) {
+        if (persons.isEmpty()) {
             return "This name is not recorded in the database";
+        } else {
+            return Utilities.buildPersonsList(persons);
         }
-        return result.toString();
     }
 
     private String modifyInsured() {
-        System.out.println("Enter the ID of the person you are looking for:");
-        String input = scanner.nextLine();
+        outputHandler.print("Enter the ID of the person you are looking for:");
+        String input = inputHandler.getInput();
         int inputId = enterNumber(input);
-        PersonI foundPerson = database.findById(inputId);
+        Person person = database.findById(inputId);
 
-        if (foundPerson == null) {
+        if (person == null) {
             return "The database does not contain the ID you entered";
         } else {
-            System.out.println(foundPerson);
+            System.out.println(person);
             String newName = enterUserName("new name");
             String newSurname = enterUserName("new surname");
             String newTel = enterPhoneNumber();
 
             database.editPerson(inputId, newName, newSurname, newTel);
-            return "The person has been modified to:\n" + foundPerson;
+            return "The person has been modified to:\n" + database.findById(inputId);
```

Comment:
instead of returning this string, it should probably by passed to the outputHandler

Reply:
It is passing to outputHandler in the method menuSelectionLoop. So probably making some of methods void would solve it? And methods used only for to-read will be still returning String into outputHandler in menuSelection?

Reply:
The caller (`menuSelectionLoop`) does not care about the return value, it just prints it using the output handler. There is no need to return the string, the method can communicate with the user directly using `outputHandler.print()`.
This is also true for all the other methods called from the main loop.